File: backend/src/Lateral.Tasks.Application/Validators/Tasks/StatusRules.cs
using Lateral.Tasks.Domain.Enums;

namespace Lateral.Tasks.Application.Validators.Tasks
{
    public static class StatusRules
    {
        public static bool IsValid(string status) =>
            Enum.TryParse<TaskItemStatus>(status, true, out _);

        public static TaskItemStatus Parse(string status) =>
            Enum.Parse<TaskItemStatus>(status, true);

        public static bool CanTransition(TaskItemStatus current, TaskItemStatus next)
        {
            if (current == TaskItemStatus.Finished) return false;

            return (current, next) switch
            {
                (TaskItemStatus.Pending, TaskItemStatus.InProgress) => true,
                (TaskItemStatus.InProgress, TaskItemStatus.Finished) => true,
                _ => false
            };
        }
    }
}

File: frontend/src/domain/tasks/statusRules.ts
import type { TaskStatus } from "./task";

export function getNextStatuses(current: TaskStatus): TaskStatus[] {
  if (current === "Pending") return ["InProgress"];
  if (current === "InProgress") return ["Finished"];
  return [];
}

export function canTransition(current: TaskStatus, next: TaskStatus): boolean {
  return getNextStatuses(current).includes(next);
}